Content.Server/Chemistry/Components/MedipenRefillerComponent.cs

```diff
@@ -0,0 +1,35 @@
+
+using Content.Shared.Chemistry;
+using Robust.Shared.Audio;
+
+namespace Content.Server.Chemistry.Components;
+
+[RegisterComponent]
+public sealed partial class MedipenRefillerComponent : Component
+{
+    [ViewVariables(VVAccess.ReadOnly)]
+    [DataField("recipes")]
+    public List<MedipenRecipePrototype> MedipenRecipes = new();
+
+    [ViewVariables(VVAccess.ReadOnly)]
+    [DataField]
+    public float CompletionTime = 20;
+
+    [ViewVariables(VVAccess.ReadOnly)]
+    public float RemainingTime = 0;
```

Comment:
TimeSpan and TimeSpan.Zero, then add customTypeSerializer: typeof(TimeOffsetSerializer)

and rename to CompleteAfter or something